js/priority-menu.js

```diff
@@ -35,94 +72,121 @@
+		var toggleButton = visibleList.parentNode.nextElementSibling.querySelector('.main-menu-more-toggle');
+
+		if ( isOverflowingNavivation( visibleList, toggleButton, container ) ) {
 
-		if (isOverflowingNavivation()) {
 			// Record the width of the list
-			breaks.push(visibleList.offsetWidth);
-			// Move item to the hidden list
-			prependElement(hiddenList, visibleList.lastChild);
+			breaks.push( visibleList.offsetWidth );
+			// Move last item to the hidden list
+			prependElement( hiddenList, ! visibleList.lastChild || null === visibleList.lastChild ? visibleList.previousElementSibling : visibleList.lastChild );
 			// Show the toggle button
-			showElement(toggleButton);
+			showElement( toggleButton );
+
 		} else {
+
 			// There is space for another item in the nav
-			if (getAvailableSpace() > breaks[breaks.length - 1]) {
+			if ( getAvailableSpace( toggleButton, container ) > breaks[breaks.length - 1] ) {
 				// Move the item to the visible list
-				visibleList.appendChild(hiddenList.firstChild.nextSibling);
+				visibleList.appendChild( hiddenList.firstChild.nextSibling );
 				breaks.pop();
 			}
 
 			// Hide the dropdown btn if hidden list is empty
 			if (breaks.length < 2) {
-				hideElement(toggleButton);
-				hideElement(hiddenList);
+				hideElement( toggleButton );
+				hideElement( hiddenList );
 			}
 		}
 
 		// Recur if the visible list is still overflowing the nav
-		if (isOverflowingNavivation()) {
-			updateNavigationMenu();
+		if ( isOverflowingNavivation( visibleList, toggleButton, container ) ) {
+			updateNavigationMenu( container );
 		}
 	}
 
-	// Event listeners
-	// Run our sub-menu function as soon as the document is `ready`
+	/**
+	 * Run our priority+ function as soon as the document is `ready`
+	 */
 	document.addEventListener( 'DOMContentLoaded', function() {
-		updateNavigationMenu();
+
+		updateNavigationMenu( navContainer );
+
+		// Also, run our priority+ function on selective refresh in the customizer
+		var hasSelectiveRefresh = (
+			'undefined' !== typeof wp &&
+			wp.customize &&
+			wp.customize.selectiveRefresh
+		);
+
+		if ( hasSelectiveRefresh ) {
+			// Force a full refresh on partial content renders to re-run updateNavigationMenu()
+			wp.customize.selectiveRefresh.bind('partial-content-rendered', function () {
+				wp.customize.preview.send('refresh');
+			});
+        }
 	});
 
+	/**
+	 * Run our priority+ function on load
+	 */
 	window.addEventListener('load', function() {
-		updateNavigationMenu();
+		updateNavigationMenu( navContainer );
 	});
 
-	window.addEventListener('resize', function() {
-		updateNavigationMenu();
-	});
+	/**
+	 * Run our priority+ function every time the window resizes
+	 */
+	var isResizing = false;
+	window.addEventListener( 'resize',
+		debounce( function() {
+			if ( isResizing ) {
+				return;
+			}
 
-	toggleButton.addEventListener('click', function() {
-		toggleElementVisibility(hiddenList);
-	});
+			isResizing = true;
+			setTimeout( function() {
```

Comment:
Why do we need setTimeout here? debounce and setTimeout are similiar functions, you can pass `150` as a second argument of debounce and I think It will have the same effect

Reply:
Ahhh, good catch—no need for setTimeout. I’ll sort that out 👍